src/dune_patch/dune_patch.ml

```diff
@@ -0,0 +1,90 @@
+open Stdune
+module Re = Dune_re
+
+include struct
+  open Dune_engine
+  module Action = Action
+  module Process = Process
+  module Utils = Utils
+end
+
+include struct
+  open Dune_lang
+  module Value = Value
+end
+
+let re =
+  Re.compile
+    (Re.non_greedy (Re.seq [ Re.str {|--- a/|}; Re.group (Re.rep1 Re.any); Re.eol ]))
+;;
+
+let files_of_patch ~dir patch =
+  Re.all re patch
+  |> List.map ~f:(fun group ->
+    let group = Re.Group.get group 1 in
+    let local_path = Path.Local.of_string group in
+    Path.append_local dir local_path)
+;;
+
+let files_of_patch_file ~dir path = Io.read_file path |> files_of_patch ~dir
+
+module Spec = struct
+  type ('path, 'target) t = 'path
+
+  let name = "patch"
+  let version = 1
+  let bimap p f _ = f p
+  let is_useful_to ~distribute:_ ~memoize = memoize
+
+  let encode (p : (_, _) t) input _ : Dune_lang.t =
+    List [ Dune_lang.atom_or_quoted_string name; input p ]
+  ;;
+
+  let action p ~(ectx : Action.Ext.context) ~(eenv : Action.Ext.env) =
+    let open Fiber.O in
+    let* () = Fiber.return () in
+    let input = Value.to_string ~dir:eenv.working_dir (Value.Path p) in
+    let patch =
+      let path = eenv.env |> Env_path.path in
```

Comment:
This is wrong. It should be be provided by the action and looked up exactly from where it's being looked up now.

Reply:
Fixed.